src/diffusers/pipelines/controlnet/pipeline_controlnet_sd_xl.py

```diff
@@ -852,11 +853,24 @@ def __call__(
             original_size, crops_coords_top_left, target_size, dtype=prompt_embeds.dtype
         )
 
+        if guess_mode and do_classifier_free_guidance:
+            # we will infer ControlNet only for the conditional batch
+            controlnet_prompt_embeds = prompt_embeds
+            controlnet_added_cond_kwargs = {"text_embeds": add_text_embeds, "time_ids": add_time_ids}
+
         if do_classifier_free_guidance:
+            # expand all inputs if we are doing classifier free guidance
             prompt_embeds = torch.cat([negative_prompt_embeds, prompt_embeds], dim=0)
             add_text_embeds = torch.cat([negative_pooled_prompt_embeds, add_text_embeds], dim=0)
             add_time_ids = torch.cat([add_time_ids, add_time_ids], dim=0)
 
+        added_cond_kwargs = {"text_embeds": add_text_embeds, "time_ids": add_time_ids}
+
+        if not guess_mode:
+            # if guess_mode is off, controlnet inputs are the same as inputs of base unet model
+            controlnet_prompt_embeds = prompt_embeds
+            controlnet_added_cond_kwargs = added_cond_kwargs
```

Comment:
How is this different from https://github.com/huggingface/diffusers/pull/4155/files#diff-905a2dbd9a8990ea1b231ad721583990f251a07e2f76138b3907729e9405417aR857-R859 ?